sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java

```diff
@@ -205,20 +209,37 @@ public ComponentDto selectOrFailByKey(DbSession session, String key) {
     return mapper(dbSession).selectAllRootsByOrganization(organizationUuid);
   }
 
-  public List<ComponentDto> selectProvisionedProjects(DbSession session, int offset, int limit, @Nullable String query) {
-    Map<String, Object> parameters = newHashMapWithExpectedSize(2);
-    addProjectQualifier(parameters);
-    addPartialQueryParameterIfNotNull(parameters, query);
-
-    return mapper(session).selectProvisionedProjects(parameters, new RowBounds(offset, limit));
+  /**
+   * Select a page of provisioned (root) components. Results are ordered by ascending name.
+   *
+   * @param dbSession
+   * @param textQuery optional text query to match component name or key
+   * @param qualifiers filter on qualifiers. Must not be null nor empty
+   * @param rowBounds pagination
+   */
+  public List<ComponentDto> selectProvisioned(DbSession dbSession, @Nullable String textQuery, Set<String> qualifiers, RowBounds rowBounds) {
+    checkArgument(!qualifiers.isEmpty(), "qualifiers must not be empty");
+    return mapper(dbSession).selectProvisioned(buildUpperLikeSql(textQuery), qualifiers, rowBounds);
   }
 
-  public int countProvisionedProjects(DbSession session, @Nullable String query) {
-    Map<String, Object> parameters = newHashMapWithExpectedSize(2);
-    addProjectQualifier(parameters);
-    addPartialQueryParameterIfNotNull(parameters, query);
+  /**
+   * Count number of provisioned (root) components.
+   *
+   * @param dbSession
+   * @param textQuery optional text query to match component name or key
+   * @param qualifiers filter on qualifiers. Must not be null nor empty
+   */
+  public int countProvisioned(DbSession dbSession, @Nullable String textQuery, Set<String> qualifiers) {
+    checkArgument(!qualifiers.isEmpty(), "qualifiers must not be empty");
+    return mapper(dbSession).countProvisioned(buildUpperLikeSql(textQuery), qualifiers);
+  }
 
-    return mapper(session).countProvisionedProjects(parameters);
+  @CheckForNull
+  private static String buildUpperLikeSql(@Nullable String textQuery) {
+    if (isBlank(textQuery)) {
+      return null;
+    }
+    return DatabaseUtils.buildLikeValue(textQuery.toUpperCase(Locale.ENGLISH), WildcardPosition.BEFORE_AND_AFTER);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') NullPointerException might be thrown as 'textQuery' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)